### PR TITLE
Autosaves: delete local auto-draft autosave after save

### LINKED ISSUE #12342: Autosaves: local `auto-draft` autosave not cleared after proper save
## Bug Description

After admin user deletes and re-installs the plugin, if a new story is added, the end user gets the "Restore from Backup" dialogue. Clicking the restore button brings an older version of the last story that was saved from the previous version of the plugin.

## Expected Behaviour

As an end user, when the plugin is deleted, there is a message that says all data will be deleted.

After the re-install, when user adds a new story, it is confusing if they are given the option to restore. And even then, the restored version is not the immediately previous version.

## Steps to Reproduce

1. Add a new story, save it as draft.
2. Add an extra page and make some edits but DO NOT SAVE
3. After about a minute, exit the story editor and open the draft story again
4. Restore Back up appears, and clicking restore shows the extra page that was added in step 2. This is expected Restore Backup workflow
5. Now disable and delete the plugin
6. Next reinstall the plugin and go to add a new story
7. You are given the Restore from backup option
8. Clicking restore gives you the draft version from step 1. The edits in step 2 above are not restored.

## Additional Context

- Plugin Version: v1.25-rc3
- WordPress Version: Wp 6.0.2
- Operating System: Win10
- Browser: Edge

### PR DESCRIPTION
**Pull request by miina**
## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Create new story
2. Wait for local autosave -> `wp_stories_autosave_story_auto-draft` gets created
3. Make proper save
4. Make more changes
5. Wait for local autosave -> `wp_stories_autosave_story_12345` gets created
6. Verify that `wp_stories_autosave_story_auto-draft` is discarded at this point

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12342

### CODE REVIEW

File: packages/story-editor/src/components/localAutoSaveHandler/localAutoSave.js [hunk 1]
```diff
@@ -97,6 +97,18 @@ function LocalAutoSave() {
     setBackup(null);
   };
 
+  const hasNewChangesTracker = useRef(false);
```
Comment (barklund) [inline packages/story-editor/src/components/localAutoSaveHandler/localAutoSave.js]: `hadNewChanges`?

**Review by barklund — COMMENTED** [review]
Seems to work great!

File: packages/story-editor/src/components/localAutoSaveHandler/localAutoSave.js (hunk 1 above)
Comment (miina) [reply packages/story-editor/src/components/localAutoSaveHandler/localAutoSave.js]: Sure, why not. Change in [2cadd17](https://github.com/GoogleForCreators/web-stories-wp/pull/12417/commits/2cadd173361b76169c305013846e6564bf0bb534)

**Review by barklund — APPROVED** [review]

**Comment by kkalarickal** [comment]
@miina @swissspidy - When I loaded the `12417 branch` within the https://stories-qa-wordpress-amp.pantheonsite.io/ , I was unable to create a new story. The browser seemed to be stuck .. perhaps its the infinite loop that you found @swissspidy 

Upon creating a new story:

Within the debugger:

So I cannot test this ticket with the changes in `12417 branch`.

Q: Could you help me get a better build made without this problem ?

**Comment by swissspidy** [comment]
Please try again now. The PR is updated now and should not have this bug anymore.

**Comment by kkalarickal** [comment]
@miina @swissspidy - I can confirm that the  `wp_stories_autosave_story_auto-draft` gets discarded when the story-in-editor is saved. It does NOT happen when the `wp_stories_autosave_story_12345 `gets created

So the sequence I observed was:
1. Create a new story and make some changes
2. Wait for `wp_stories_autosave_story_auto-draft`  to be created
3. Make a proper save

NOW the `wp_stories_autosave_story_auto-draft` is discarded

4. Make some more changes
5. `wp_stories_autosave_story_12345 `gets created after some time

I was expecting the `wp_stories_autosave_story_auto-draft`  to be discarded after step 5. Instead is happens after step 3.

Q: Could you confirm this sequence is ok ?

**Comment by kkalarickal** [comment]
In addition, I would like to flag a regression:

When the `wp_stories_autosave_story_auto-draft` is used to restore a story, the new story created is BLANK !

Steps
1. Create a new story with some basic elements (its important to have something in the story to distinguish it from a blank story)
2. Wait for `wp_stories_autosave_story_auto-draft` to be created
3. Confirm that the elements are in the session-storage
4. DO NOT save the story, and go back to Stories > Dashboard, purposely clicking "Leave Site" button
5. Click on "Create a new story" once again
6. The plugin will detect the `wp_stories_autosave_story_auto-draft` being present and ask if you want to restore from backup
7. Click restore

ISSUE: A blank story is created
EXPECTED: The half baked story from step 1 should be restored.

This is working as expected on `main branch` but yields the blank story regression on the `12417 branch`

Cc: @miina

**Comment by miina** [comment]
> I was expecting the wp_stories_autosave_story_auto-draft to be discarded after step 5. Instead is happens after step 3.

@kkalarickal Yes, that's ok, thanks for checking. Since the saved version is also stored in the database then there's no need for an older local backup.

**Comment by miina** [comment]
> In addition, I would like to flag a regression:

Confirmed the issue and should be fixed now as well, could you test again?

**Comment by kkalarickal** [comment]
Confirmed that the `wp_stories_autosave_story_auto-draft` is discarded when there is a proper save 
Also confirmed that the restore from `auto-draft` is working as expected as is the restore from a session copy after manual save+edits.

There are some more scenarios around take over and auto-saved session objects ... but those are unrelated to this ticket. I will file them separately.

@miina @swissspidy - this ticket can be merged.

**Pull request merged**